elasticsearch_dsl/search_base.py

```diff
@@ -619,8 +623,16 @@ def source(self, fields=None, **kwargs):
         if fields and kwargs:
             raise ValueError("You cannot specify fields and kwargs at the same time.")
 
+        def ensure_strings(fields):
+            if isinstance(fields, list):
+                return [str(f) for f in fields]
+            elif isinstance(fields, dict):
+                return {k: ensure_strings(v) for k, v in fields.items()}
+            else:
+                return str(fields)
```

Comment:
Why didn't we need this before?

Reply:
Because using class attributes in sort is a new feature I'm introducing with this change. In the current version only string representations of fields are supported and accessing a class attribute in the current version returns `AttributeError`.

I like the idea of using the class attributes to reference to fields, like SQLAlchemy does, and that means that we may get a mix of strings and the new `InstrumentedAttribute` instances whenever field names are referenced.